### PR TITLE
Removing .NET Core 3.1, Adding .NET 8

### PR DESCRIPTION
**Pull request by JasonBock**
## Description

Updating TFMs, mostly to remove .NET Core 3.1 as it is no longer supported.

## Type of change:

4. Breaking change (fix or feature that could cause existing functionality to not work as expected)

### CODE REVIEW

File: RockLib.Configuration/README.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 Defines a static `Config` class as a general replacement for the old .NET Framework `ConfigurationManager` class.
 
+Note: The 3.2.0 release of this library will be the final version with upgrades and changes. Bug fixes will continue to be released as needed. We strongly encourage developers to use standard .NET configuration libraries directly like `Microsoft.Extensions.Configuration` in place of `RockLib.Configuration`.
```
Comment (aatchison14) [inline RockLib.Configuration/README.md]:
```suggestion
Note: The 4.0.0 release of this library will be the final version with upgrades and changes. Bug fixes will continue to be released as needed. We strongly encourage developers to use standard .NET configuration libraries directly like `Microsoft.Extensions.Configuration` in place of `RockLib.Configuration`.
```

4.0.0?

File: Tests/RockLib.Configuration.AppSettingsConfigTests/Directory.Build.props [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
 		<Nullable>enable</Nullable>
-		<TargetFrameworks>net6.0;netcoreapp3.1;net48</TargetFrameworks>
+		<TargetFrameworks>net48;net6.0;net8.0</TargetFrameworks>
 		<NoWarn>NU1603,NU1701</NoWarn>
 		<TreatWarningsAsErrors>true</TreatWarningsAsErrors>
 	</PropertyGroup>
-	<ItemGroup Condition=" '$(TargetFramework)' == 'net48' or '$(TargetFramework)' == 'netcoreapp3.1'">
-		<PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="6.0.0">
+	<ItemGroup Condition=" '$(TargetFramework)' == 'net48'">
+		<PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="[6, 8)">
```
Comment (aatchison14) [inline Tests/RockLib.Configuration.AppSettingsConfigTests/Directory.Build.props]: why does this begin with `[` and end with `)`?

File: Tests/RockLib.Configuration.CustomConfigTests/Directory.Build.props [hunk 3]
```diff
@@ -1,18 +1,18 @@
...
 		<Nullable>enable</Nullable>
-		<TargetFrameworks>net6.0;netcoreapp3.1;net48</TargetFrameworks>
+		<TargetFrameworks>net48;net6.0;net8.0</TargetFrameworks>
 		<NoWarn>NU1603,NU1701</NoWarn>
 		<TreatWarningsAsErrors>true</TreatWarningsAsErrors>
 	</PropertyGroup>
-	<ItemGroup Condition=" '$(TargetFramework)' == 'net48' or '$(TargetFramework)' == 'netcoreapp3.1'">
-		<PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="6.0.0">
+	<ItemGroup Condition=" '$(TargetFramework)' == 'net48'">
+		<PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="[6, 8)">
```
Comment (aatchison14) [inline Tests/RockLib.Configuration.CustomConfigTests/Directory.Build.props]: same here with the `[` and `)`

File: Tests/RockLib.Configuration.AppSettingsConfigTests/Directory.Build.props (hunk 2 above)
Comment (JasonBock) [reply Tests/RockLib.Configuration.AppSettingsConfigTests/Directory.Build.props]: It's a version range: https://learn.microsoft.com/en-us/nuget/concepts/package-versioning#version-ranges

File: RockLib.Configuration/Directory.Build.props [hunk 4]
```diff
@@ -1,18 +1,18 @@
...
 		<Nullable>enable</Nullable>
-		<TargetFrameworks>net6.0;netcoreapp3.1;net48</TargetFrameworks>
+		<TargetFrameworks>net48;net6.0;net8.0</TargetFrameworks>
 		<NoWarn>NU1603,NU1701</NoWarn>
 		<TreatWarningsAsErrors>true</TreatWarningsAsErrors>
 	</PropertyGroup>
-	<ItemGroup Condition=" '$(TargetFramework)' == 'net48' or '$(TargetFramework)' == 'netcoreapp3.1'">
-		<PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="6.0.0">
+	<ItemGroup Condition=" '$(TargetFramework)' == 'net48'">
+		<PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="[6, 8)">
```
Comment (aatchison14) [inline RockLib.Configuration/Directory.Build.props]: and here

File: Tests/RockLib.Configuration.AppSettingsConfigTests/Directory.Build.props (hunk 2 above)
Comment (aatchison14) [reply Tests/RockLib.Configuration.AppSettingsConfigTests/Directory.Build.props]: ah, inclusivity....that makes sense. after the 4th one I saw, I knew I was missing something lol. Thanks!

**Review by aatchison14 — DISMISSED** [review]

**Review by pjwilliams2 — APPROVED** [review]

**Review by NeilSchiefer — APPROVED** [review]

**Review by NeilSchiefer — DISMISSED** [review]

**Review by NeilSchiefer — APPROVED** [review]

File: Tests/RockLib.Configuration.MessagingProvider.Tests/RockLib.Configuration.MessagingProvider.Tests.csproj [hunk 5]
```diff
@@ -11,16 +12,16 @@
...
-		<PackageReference Include="FluentAssertions" Version="6.5.1" />
-		<PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.1.0" />
-		<PackageReference Include="Moq" Version="4.17.2" />
-		<PackageReference Include="RockLib.UniversalMemberAccessor" Version="2.0.0" />
-		<PackageReference Include="xunit" Version="2.4.1" />
-		<PackageReference Include="xunit.runner.visualstudio" Version="2.4.3">
+		<PackageReference Include="FluentAssertions" Version="6.12.0" />
+		<PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.8.0" />
+		<PackageReference Include="Moq" Version="4.20.70" />
+		<!--<PackageReference Include="RockLib.UniversalMemberAccessor" Version="2.0.0" />-->
```
Comment (NeilSchiefer) [inline Tests/RockLib.Configuration.MessagingProvider.Tests/RockLib.Configuration.MessagingProvider.Tests.csproj]: Oh wait lets just remove this.:)

File: RockLib.Configuration/CHANGELOG.md [hunk 6]
```diff
@@ -5,6 +5,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## 4.0.0-alpha.1 - Not Yet Released
+
+#### Changed
+- Removed .NET Core 3.1 as a TFM and added .NET 8. Supported targets are now .NET 4.8, .NET 6, and .NET 8.
+- Removed RockLib.Immutable as a dependency.
```
Comment (kristycurrier) [inline RockLib.Configuration/CHANGELOG.md]: Might want to add a little more detail about removing it and adding functionality (classes) to this library.

**Review by NeilSchiefer — APPROVED** [review]

**Review by aatchison14 — APPROVED** [review]

**Pull request merged**